### DISCUSSION TITLE
Confusion with Environment tag?

### DISCUSSION THREAD

**Discussion by kevin--** [getsentry/support#41] [Support] [answered]
Hi all,

My client does not set environment (C++ SDK `sentry_options_set_environment`) because it does not know the environment. However once the release is vetted by QA and our beta users, we then deploy it to "Production" for example
```
sentry-cli releases deploys "$VERSION" new -e "Production"
```
However when events come in on a deployed version, and we filter on Production, the events are hidden.

How is this supposed to work? Must the Event and the Deploy "Production" tag match? 

The net result I want is to only see crashes that occur in Deployed Production releases.

**Comment by Kobby-Bawuah** [comment, marked as answer]
Hello @kevin-- 

I can see you sent a ticket to support about this issue on the 21st of July. It seems it was solved. Let me know if this is still a concern.

**Reply by kevin--** [reply]
Yes -- to clarify for anyone else having this issue.
- the Environment tag on events is not impacted by the Environment tag of Deploys
- Therefore, if you want your events to show up in the "Production" environment for example, to match your Deploy, you have to tag them that way at compile time (C++ Native SDK)

Still feels like a deficiency in the platform to be honest. I don't understand why you would want events to have an Environment that is not associated with their deployment?

We have to restrict the use of "Production" events to only be for release-candidate builds only, and then know that we must ignore any "Production" events not associated with a "Production" deploy. (for example a build that went to Beta and then was rejected may have generated "Production" events, but are not actually associated with customers in the wild)
